node/src/test/scala/com/wavesplatform/http/AddressRouteSpec.scala

```diff
@@ -218,24 +228,42 @@ class AddressRouteSpec
           .anyNumberOfTimes()
     }
 
-    val regex = """^Some.*$"""
+    for (regex <- regexps) {
+      Get(routePath(s"""/data/${allAddresses(1)}?matches=$regex""")) ~> route ~> check {
+        val kvs = responseAs[JsArray].value.map { json =>
+          ((json \ "key").as[String], (json \ "value").as[String])
+        }.toList.sortBy(_._1)
 
-    Get(routePath(s"""/data/${allAddresses(1)}?matches=$regex""")) ~> route ~> check {
-      val kvs = responseAs[JsArray].value.map { json =>
-        ((json \ "key").as[String], (json \ "value").as[String])
-      }.toList
-
-      assert(kvs forall { case (k, v) => k.startsWith("Some") && testData(k) == v })
+        val regexPattern = regex.r.pattern
+        kvs shouldEqual testData.filter(k => regexPattern.matcher(k._1).matches()).toSeq.sortBy(_._1)
+      }
     }
 
-    val urlEncRegex = """%5ESome.%2A%24"""
+    for (regex <- regexps.map(rgx => URLEncoder.encode(rgx, "UTF-8"))) {
+      Get(routePath(s"""/data/${allAddresses(1)}?matches=$regex""")) ~> route ~> check {
+        val kvs = responseAs[JsArray].value.map {
+          json => ((json \ "key").as[String], (json \ "value").as[String])
+        }.toList.sortBy(_._1)
 
-    Get(routePath(s"""/data/${allAddresses(1)}?matches=$urlEncRegex""")) ~> route ~> check {
-      val kvs = responseAs[JsArray].value.map { json =>
-        ((json \ "key").as[String], (json \ "value").as[String])
-      }.toList
+        val regexPattern = URLDecoder.decode(regex, "UTF-8").r.pattern
+        kvs shouldEqual testData.filter(k => regexPattern.matcher(k._1).matches()).toSeq.sortBy(_._1)
+      }
+    }
 
-      assert(kvs forall { case (k, v) => k.startsWith("Some") && testData(k) == v })
+    val invalidRegexps = List("[a-z", "([a-z]{0}", "[a-z]{0", "[a-z]{,5}")
+    for (regex <- invalidRegexps) {
+      try {
+        Get(routePath(s"""/data/${allAddresses(1)}?matches=$regex""")) ~> route ~> check {
+          responseAs[JsArray].value.map { json =>
+            ((json \ "key").as[String], (json \ "value").as[String])
+          }
+        }
+        fail("RegexCompiler didn't throw expected error")
+      } catch {
+        case err: Throwable => if (!err.getMessage.contains("Cannot compile regex")) {
+          throw err
```

Comment:
if an exception is expected here, then it is probably better to use `assertThrows` of ScalaTest

Reply:
as i understand assertThrows need concrete type of exception but here need to check only exception message
i guess better solution is to check message in response right after request without throwing added exceptions